Source/Core/Core/PowerPC/JitArm64/JitArm64_Branch.cpp

```diff
@@ -252,18 +212,15 @@ void JitArm64::bcx(UGeckoInstruction inst)
     gpr.Flush(FlushMode::MaintainState, WB);
     fpr.Flush(FlushMode::MaintainState, ARM64Reg::INVALID_REG);
 
-    if (IsDebuggingEnabled())
+    if (IsBranchWatchEnabled())
     {
-      ARM64Reg bw_reg_a, bw_reg_b;
-      // WC is only allocated when WA is needed for WriteExit and cannot be clobbered.
-      if (WC == ARM64Reg::INVALID_REG)
-        bw_reg_a = WA, bw_reg_b = WB;
-      else
-        bw_reg_a = WB, bw_reg_b = WC;
-      const BitSet32 gpr_caller_save =
-          gpr.GetCallerSavedUsed() & ~BitSet32{DecodeReg(bw_reg_a), DecodeReg(bw_reg_b)};
-      WriteBranchWatch<true>(js.compilerPC, js.op->branchTo, inst, bw_reg_a, bw_reg_b,
-                             gpr_caller_save, fpr.GetCallerSavedUsed());
+      BitSet32 gpr_caller_save = gpr.GetCallerSavedUsed() & ~BitSet32{DecodeReg(WB)};
+      // WA can only be clobbered if it's not needed for WriteExit.
+      if (js.op->branchIsIdleLoop)
```

Comment:
WA is only used by WriteExit if `inst.LK && !js.op->branchIsIdleLoop`. (The `LK` check is inside `WriteExit`.)

Reply:
If `!(inst.lk)` then WA and WB are the same register: because WB is never saved, then technically WA isn't saved in this case either.
Granted, this is not obvious at first glance... do I add a redundant check for `inst.lk` for clarity, or should I explain it in the comment?